RxSwift/Observables/Breakpoint.swift

```diff
@@ -0,0 +1,182 @@
+     - returns: The source sequence that raises a debugger signal when one of the provided closures returns `true`.
+     */
+    public func breakpoint(
+        onNext: @escaping (Element) throws -> Bool = { _ in false },
+        afterNext: @escaping (Element) throws -> Bool = { _ in false },
+        onError: @escaping (Swift.Error) throws -> Bool = { _ in false },
+        afterError: @escaping (Swift.Error) throws -> Bool = { _ in false },
+        onCompleted: @escaping @autoclosure () throws -> Bool = false,
+        afterCompleted: @escaping @autoclosure () throws -> Bool = false,
+        onSubscribe: @escaping @autoclosure () -> Bool = false,
+        onSubscribed: @escaping @autoclosure () -> Bool = false,
+        onDispose: @escaping @autoclosure () -> Bool = false
+    ) -> Observable<Element> {
+        return Breakpoint(
+            source: self.asObservable(),
+            eventPredicate: {
+                switch $0 {
+                case let .next(element):
+                    return try onNext(element)
+                case let .error(error):
+                    return try onError(error)
+                case .completed:
+                    return try onCompleted()
+                }
+            },
+            afterEventPredicate:  {
+                switch $0 {
+                case let .next(element):
+                    return try afterNext(element)
+                case let .error(error):
+                    return try afterError(error)
+                case .completed:
+                    return try afterCompleted()
+                }
+            },
+            onSubscribe: onSubscribe(),
+            onSubscribed: onSubscribed(),
+            onDispose: onDispose()
+        )
+    }
+
+    /**
+      Raises a debugger signal upon receiving an error.
+      When the upstream Observable fails with an error, this operator raises the `SIGTRAP` signal, which stops the process in the debugger.
+      Otherwise, this Observable passes through values and completions as-is.
+      - seealso: [breakpointOnError operator on Combine](https://developer.apple.com/documentation/combine/publisher/3204689-breakpointonerror)
+      - returns: An Observable that raises a debugger signal upon receiving a failure.
+      */
+    public func breakpointOnError() -> Observable<Element> {
+        return Breakpoint(
+            source: self.asObservable(),
+            eventPredicate: {
+                switch $0 {
+                case .error:
+                    return true
+                default:
+                    return false
+                }
+            },
+            afterEventPredicate: { _ in false }
+        )
+    }
+}
+
+#if DEBUG
+private func raiseBreakpoint() {
+    raise(SIGINT)
```

Comment:
You can test this by setting a signal event handler, something like 

```swift
var trappedSignal = false
signal(SIGINT) { _ in
  trappedSignal = true
}

XCTAssert...
```

At least IIRC... worth trying :)

Reply:
Here's another option: 
https://developer.apple.com/documentation/dispatch/dispatchsource/2300045-makesignalsource

Reply:
The problem with both solutions - Xcode debugger handles signals and prevent application handlers to catch them. 
I can fix that with `process handle SIGTRAP -p true`. But I need to do it manually every time. I can also disable debugger for the whole test target, but I doubt that it's a correct solution.

Reply:
Yeah I understand. that's a bit tricky, I would want at least some sort of test here... with xcodebuild I guess this would work though, rihgt?